test/distributed/_tensor/test_math_ops.py

```diff
@@ -25,8 +25,13 @@ def linear_op_reductions(self, op_str):
         device_mesh = self.build_device_mesh()
         shard_spec = [Shard(0)]
 
-        tensor = torch.randn(12, 8, 8)
-        dtensor = distribute_tensor(tensor, device_mesh, shard_spec)
+        if op_str == "any":
+            # test out a bool tensor for any op
+            tensor = torch.randn(8, 2) < 0
```

Comment:
Alternatively you can use `torch.randn(8, 2).bool()` here.

Also I think for `aten.all` the conversion is implicit? so we probably don't need this if-else anyway. I suggest we either put `all` and `any` together into this branch, or remove this branch.

Reply:
The only number that would be evaluated to be False is 0 so most likely we might not have any False in `torch.randn(8, 2).bool()`. Keeping the < 0 here so we would have some False in the bool tensor. 

I'll move "all" to the branch as well.

Reply:
Actually, I realized the result for `all` might be wrong, but the test didn't catch it since we are testing it on `tensor = torch.randn(12, 8, 8) `, which will likely have everything to be True. 

Added a TODO to check `all` correctness but want to land this to unblock something else first.